src/App.tsx

```diff
@@ -1,24 +1,286 @@
+
+      setTodos((prevTodos) => prevTodos.map((todo) => (
+        todo.id === todoToUptade.id ? updatedTodo : todo
+      )));
+    } catch (error) {
+      setVisibleError(NewError.Update);
+    } finally {
+      setUpdatingTodoId(null);
+    }
+  }, [todos]);
+
+  const handleUpdateAllTodos = useCallback(async () => {
+    setIsUpdatingEveryStatus(true);
+    try {
+      await Promise.all(
+        todos.map((todo) => updateTodo({
+          ...todo,
+          completed: !isEveryTotoCompleted,
+        })),
+      );
+
+      setTodos((prevTodos) => prevTodos.map((todo) => ({
+        ...todo,
+        completed: !isEveryTotoCompleted,
+      })));
+    } catch (error) {
+      setVisibleError(NewError.Update);
+    } finally {
+      setIsUpdatingEveryStatus(false);
+    }
+  }, [todos]);
+
+  const handleUpdateTodoTitle = useCallback(async (
+    todoToUpdate: Todo,
+    newTitle: string,
+  ) => {
+    setUpdatingTodoId(todoToUpdate.id);
+
+    try {
+      const updatedTodo = await updateTodo({
+        ...todoToUpdate,
+        title: newTitle,
+      });
+
+      setTodos((prevTodos) => prevTodos.map((todo) => (
+        todo.id === todoToUpdate.id ? updatedTodo : todo
+      )));
+    } catch (error) {
+      setVisibleError(NewError.Update);
+    } finally {
+      setUpdatingTodoId(null);
+    }
+  }, [todos]);
+
+  if (visibleError) {
+    setTimeout(() => {
+      setVisibleError(null);
+    }, 3000);
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {isTodosNoEmpty && (
+            <button
+              type="button"
+              className={classNames(
+                'todoapp__toggle-all', {
+                  active: isEveryTotoCompleted,
+                },
+              )}
+              onClick={handleUpdateAllTodos}
+            />
+          )}
+
+          <form onSubmit={handleSubmit}>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={query}
+              onChange={(event) => handleQueryChange(event.target.value)}
+            />
+          </form>
+        </header>
+
+        <TodoList
+          todos={visibleTodos}
+          tempTodo={tempTodo}
+          updatingTodoId={updatingTodoId}
+          isRemovingCompleted={isRemovingCompleted}
+          isUpdatingEveryStatus={isUpdatingEveryStatus}
+          isEveryTotoCompleted={isEveryTotoCompleted}
```

Comment:
```suggestion
          isRemovingCompleted={isRemovingCompleted}
          isUpdatingStatus={isUpdatingEveryStatus}
          isAllTodoCompleted={isEveryTotoCompleted} // you can do it in the list
```

Reply:
I use this variable in 2 components (Heder and TodoList).

Reply:
Ok, agree with you